scripts/get_indexfiles.py

```diff
@@ -97,17 +121,32 @@ def main():
         cluster_idx = Path(clusters_dir) / f'{args.collection}_cluster.idx'
         if not cluster_idx.is_file():
             logging.info(f'Downloading cluster index for {args.collection}...')
+            logging.debug(base_url + 'cluster.idx')
             urllib.request.urlretrieve(base_url + 'cluster.idx', cluster_idx)
 
-        # Then, get the files and byte ranges that correspond to the TLD
-        clusters = find_tld_in_index(args.tld, cluster_idx)
+        # Then, get the files and byte ranges that correspond to the query:
+        clusters = []
+        for pattern_list in patterns_as_lists:
+            current_clusters = find_pattern_in_index(pattern_list, cluster_idx)
+            clusters += current_clusters
+        clusters = list(set(clusters))  # We remove duplicates
```

Comment:
What about
```Python
clusters = set()
for pattern_list in patterns_as_lists:
    clusters.update(find_pattern_in_index(pattern_list, cluster_idx))
return sorted(clusters, key=lambda cluster: (cluster.file_name, cluster.offset))
```
all in a separate function? No need to keep a long list in memory and then convert it back-and-forth between `set` and `list`.

Reply:
Good idea!.